runtime/ops/fs.rs

```diff
@@ -1252,6 +1252,9 @@ fn op_symlink_sync(
   let oldpath = PathBuf::from(&args.oldpath);
   let newpath = PathBuf::from(&args.newpath);
 
+  state.borrow_mut::<Permissions>().read.check(&oldpath)?;
```

Comment:
* This is not correct. Symlinks are resolved relative to their own location; this assumes that they're resolved relative to CWD.
* A sandbox escape is still possible by creating a symlink to "../secret" in "./allowed1/allowed2/symlink_here", and then moving the symlink from "allowed1/allowed2" to "allowed1".

Reply:
> This is not correct. Symlinks are resolved relative to their own location; this assumes that they're resolved relative to CWD.

Can you elaborate on this one?

> A sandbox escape is still possible by creating a symlink to "../secret" in "./allowed1/allowed2/symlink_here", and then moving the symlink from "allowed1/allowed2" to "allowed1".

Yes, @ry and I both agree on that and count it as separate issue tracked in #2318 and #9607.

The intent of this PR was to fix the immediate bug of not checking one permission when creating a symlink.

Reply:
> > This is not correct. Symlinks are resolved relative to their own location; this assumes that they're resolved relative to CWD.
> 
> Can you elaborate on this one?

If you create a symlink named `some/dir/over/there/mylink` with content `../target`, the symlink will point at `some/link/over/target`. So the permissions should be checked for these two paths:
  * `some/dir/over/there/mylink` (`newpath`)
  * `some/link/over/target` (`join(dirname(newpath), oldpath)`)

It currently checks permissions for:
  * `some/dir/over/there/mylink` (`newpath` - correct)
  * `../target` (`oldpath` - incorrect)